src/ui/public/agg_types/metrics/lib/parent_pipeline_agg_helper.js

```diff
@@ -0,0 +1,61 @@
+import metricAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+import { parentPipelineAggController } from './lib/parent_pipeline_agg_controller';
+import { parentPipelineAggWritter } from './lib/parent_pipeline_agg_writter';
+
+const ParentPipelineAggHelperProvider = function (Private) {
+
+  const AggConfig = Private(VisAggConfigProvider);
+  const Schemas = Private(VisSchemasProvider);
+
+  const metricAggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+  const metricAggSchema = (new Schemas([
+    {
+      group: 'none',
+      name: 'metricAgg',
+      title: 'Metric Agg',
+      aggFilter: metricAggFilter
+    }
+  ])).all[0];
+
+  return {
+    params: function () {
+      return [
+        {
+          name: 'customMetric',
+          type: AggConfig,
+          default: null,
+          serialize: function (customMetric) {
+            return customMetric.toJSON();
+          },
+          deserialize: function (state, agg) {
+            return this.makeAgg(agg, state);
+          },
```

Comment:
Up to you but you can compress these 6 lines to two via:
```
serialize: (customMetric) => customMetric.toJSON(),
deserialize: (state, agg) => this.makeAgg(agg, state),
```
One minor note, I like to keep the order of params somewhat consistent and here we are switching the order of agg and state, but I'm guessing this is part or the interface and wouldn't be a simple thing to change, so fine as is.